crates/factor-outbound-pg/src/client.rs

```diff
@@ -0,0 +1,75 @@
+use native_tls::TlsConnector;
+use postgres_native_tls::MakeTlsConnector;
+use spin_world::async_trait;
+use tokio_postgres::{config::SslMode, types::ToSql, Error, Row, ToStatement};
+use tokio_postgres::{Client as PgClient, NoTls, Socket};
+
+#[async_trait]
+pub trait Client {
+    async fn build_client(address: &str) -> anyhow::Result<Self>
+    where
+        Self: Sized;
+
+    async fn execute<T>(&self, query: &T, params: &[&(dyn ToSql + Sync)]) -> Result<u64, Error>
+    where
+        T: ?Sized + ToStatement + Sync + Send;
+
+    async fn query<T>(&self, query: &T, params: &[&(dyn ToSql + Sync)]) -> Result<Vec<Row>, Error>
+    where
+        T: ?Sized + ToStatement + Sync + Send;
+}
+
+#[async_trait]
+impl Client for PgClient {
+    async fn build_client(address: &str) -> anyhow::Result<Self>
+    where
+        Self: Sized,
+    {
+        let config = address.parse::<tokio_postgres::Config>()?;
+
+        tracing::debug!("Build new connection: {}", address);
+
+        if config.get_ssl_mode() == SslMode::Disable {
+            async move {
+                let (client, connection) = config.connect(NoTls).await?;
+                spawn(connection);
+                Ok(client)
+            }
+            .await
+        } else {
+            async move {
+                let builder = TlsConnector::builder();
+                let connector = MakeTlsConnector::new(builder.build()?);
+                let (client, connection) = config.connect(connector).await?;
+                spawn(connection);
+                Ok(client)
+            }
+            .await
+        }
```

Comment:
Nit: this removes a bit of redundant boiler plate and makes the code a bit easier to read IMO.
```suggestion
        Ok(async move {
            if config.get_ssl_mode() == SslMode::Disable {
                let (client, connection) = config.connect(NoTls).await?;
                spawn(connection);
                client
            } else {
                let builder = TlsConnector::builder();
                let connector = MakeTlsConnector::new(builder.build()?);
                let (client, connection) = config.connect(connector).await?;
                spawn(connection);
                client
            }
        }.await)
```